src/elexmodel/handlers/data/Estimandizer.py

```diff
@@ -0,0 +1,75 @@
+from numpy import nan
+
+
+class EstimandException(Exception):
+    pass
+
+
+RESULTS_PREFIX = "results_"
+BASELINE_PREFIX = "baseline_"
+
+
+class Estimandizer:
+    """
+    Estimandizer. Generate estimands explicitly.
+    """
+
+    def add_estimand_results(self, data_df, estimands, historical):
+        columns_to_return = []
+        for estimand in estimands:
+            results_col = f"{RESULTS_PREFIX}{estimand}"
+            if results_col not in data_df.columns:
+                # will raise a KeyError if a function with the same name as `estimand` doesn't exist
+                try:
+                    data_df = globals()[estimand](data_df, RESULTS_PREFIX)
+                except KeyError as e:
+                    if historical:
+                        data_df[results_col] = nan
+                    else:
+                        raise e
+            columns_to_return.append(results_col)
+
+        results_column_names = [x for x in data_df.columns if x.startswith(RESULTS_PREFIX)]
+        # If this is not a historical run, then this is a live election
+        # so we are expecting that there will be actual results data
+        if not historical and len(results_column_names) == 0:
+            raise EstimandException("This is not a test election, it is missing results data")
+
+        return data_df, columns_to_return
+
+    def add_estimand_baselines(self, data_df, estimand_baselines, historical, include_results_estimand=False):
+        # if we are in a historical election we are only reading preprocessed data to get
+        # the historical election results of the currently reporting units.
+        # so we don't care about the total voters or the baseline election.
+
+        for estimand, pointer in estimand_baselines.items():
+            if pointer is None:
+                # when we are creating a new estimand
+                pointer = estimand
+
+            baseline_col = f"{BASELINE_PREFIX}{pointer}"
+
+            if baseline_col not in data_df.columns:
+                data_df = globals()[estimand](data_df, BASELINE_PREFIX)
+
+            if not historical:
+                data_df[f"last_election_results_{estimand}"] = data_df[baseline_col].copy() + 1
+
+        if include_results_estimand:
```

Comment:
similarly here do you mind adding a column? Something like: in the historical election results case we use the preprocessed data to get the current results too, but the preprocessed data only calls add_estimand_baseline, but in that case we want to add the results from the historical election also.

Reply:
Done!  Please check my comments and let me know if they make sense (or not 😅 ).  Thanks!